{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.10.12","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"},"kaggle":{"accelerator":"none","dataSources":[],"dockerImageVersionId":30839,"isInternetEnabled":false,"language":"python","sourceType":"notebook","isGpuEnabled":false}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"# This Python 3 environment comes with many helpful analytics libraries installed\n# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python\n# For example, here's several helpful packages to load\n\nimport numpy as np # linear algebra\nimport pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n\n# Input data files are available in the read-only \"../input/\" directory\n# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory\n\nimport os\nfor dirname, _, filenames in os.walk('/kaggle/input'):\n    for filename in filenames:\n        print(os.path.join(dirname, filename))\n\n# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using \"Save & Run All\" \n# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session","metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","trusted":true,"execution":{"iopub.status.busy":"2025-01-20T17:14:23.786204Z","iopub.execute_input":"2025-01-20T17:14:23.78654Z","iopub.status.idle":"2025-01-20T17:14:24.167593Z","shell.execute_reply.started":"2025-01-20T17:14:23.786513Z","shell.execute_reply":"2025-01-20T17:14:24.166491Z"}},"outputs":[],"execution_count":1},{"cell_type":"code","source":"!sudo apt-get update && sudo apt-get install -y python3.11 python3.11-dev python3.11-venv","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-01-20T17:14:24.168836Z","iopub.execute_input":"2025-01-20T17:14:24.169295Z"}},"outputs":[{"name":"stdout","text":"Ign:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\nIgn:2 http://security.ubuntu.com/ubuntu jammy-security InRelease               \nIgn:3 https://r2u.stat.illinois.edu/ubuntu jammy InRelease                     \nIgn:4 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease   \nIgn:5 http://archive.ubuntu.com/ubuntu jammy InRelease                         \nIgn:6 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease     \nIgn:7 http://archive.ubuntu.com/ubuntu jammy-updates InRelease                 \nIgn:8 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\nIgn:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\nIgn:2 http://security.ubuntu.com/ubuntu jammy-security InRelease               \nIgn:6 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease     \nIgn:3 https://r2u.stat.illinois.edu/ubuntu jammy InRelease                     \nIgn:9 http://archive.ubuntu.com/ubuntu jammy-backports InRelease               \nIgn:10 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease   \nIgn:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\nIgn:3 https://r2u.stat.illinois.edu/ubuntu jammy InRelease                     \nIgn:2 http://security.ubuntu.com/ubuntu jammy-security InRelease               \nIgn:6 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease     \nIgn:5 http://archive.ubuntu.com/ubuntu jammy InRelease                         \nIgn:4 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease   \nErr:6 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease     \n  Temporary failure resolving 'cloud.r-project.org'\nErr:3 https://r2u.stat.illinois.edu/ubuntu jammy InRelease                     \n  Temporary failure resolving 'r2u.stat.illinois.edu'\nErr:2 http://security.ubuntu.com/ubuntu jammy-security InRelease               \n  Temporary failure resolving 'security.ubuntu.com'\nErr:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n  Temporary failure resolving 'developer.download.nvidia.com'\nIgn:7 http://archive.ubuntu.com/ubuntu jammy-updates InRelease                 \nIgn:8 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\nIgn:9 http://archive.ubuntu.com/ubuntu jammy-backports InRelease              \nIgn:10 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease  \nIgn:5 http://archive.ubuntu.com/ubuntu jammy InRelease                        \nIgn:4 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease  \nIgn:7 http://archive.ubuntu.com/ubuntu jammy-updates InRelease                \nIgn:8 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\nIgn:10 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease  \nIgn:9 http://archive.ubuntu.com/ubuntu jammy-backports InRelease              \n0% [Connecting to archive.ubuntu.com] [Connecting to ppa.launchpadcontent.net]","output_type":"stream"}],"execution_count":null},{"cell_type":"code","source":"!sudo apt-get update && sudo apt-get install -y python3.11 python3.11 python3.11-venv\n!sudo rm /usr/bin/python3\n!sudo ln -s /usr/bin/python3.11 /usr/bin/python3\n!sudo ln -sf /usr/bin/python3.11 /usr/bin/python\n\n!python --version\n!python3 --version","metadata":{"trusted":true},"outputs":[],"execution_count":null}]}